sql/core/src/main/scala/org/apache/spark/sql/scripting/SqlScriptingExecutionNode.scala

```diff
@@ -0,0 +1,135 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.scripting
+
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+import org.apache.spark.sql.catalyst.trees.{Origin, WithOrigin}
+
+/**
+ * Trait for all SQL scripting execution nodes used during interpretation phase.
+ */
+sealed trait CompoundStatementExec extends Logging {
+  /**
+   * Whether the statement originates from the SQL script or is created during the interpretation.
+   * Example: DropVariable statements are automatically created at the end of each compound.
```

Comment:
Anything other than drop var are we planning to add in the future? drop temp table/view/functions?

Reply:
Definitely, we will have other temporary stuff to drop in the future. For now, we made a really simple change like this, so our scripts are functional for testing, etc.

Follow-up PRs will first introduce labels for compound bodies (both implicit and explicit) first, which is a requirement for further work/improvements in the variables area - like introducing variable shadowing and similar. Once the labels are introduced, we will work on variable improvements - this will probably include some type of a context (which might later be used for a ton of other stuff - call stack tracking once we introduce procedures, tracking info for some other statements like BREAK/CONTINUE, etc). Then, we will exactly know how we want to handle these droppings - it probably won't be as explicit as in this PR, but rather include some interaction with the context. Then, we will figure out exactly whether we need this `isInternal` property down the road or not.

Reply:
It is questionable whether temp table/view/function should be tied to a compound scope (probably not) like variables. They will continue to live on session level.